lab3/client/src/views/generator.js

```diff
@@ -0,0 +1,60 @@
+import constants from '../constants.js';
+
+export default function (store) {
+	return class GeneratorComponent extends window.HTMLElement {
+		constructor () {
+			super();
+			this.store = store;
+
+			// TODO: render generator initial view
+			this.innerHTML = this.render();
+
+			// TODO: subscribe to store on change event
+			this.onStateChange = this.handleStateChange.bind(this);
+
+			// TODO: add click event
+			this.addEventListener('click', () => {
```

Comment:
This click event is binding to wrong element.

You should bind the event to the "button" below not the whole element.